content/stub.completion-ui.js

```diff
@@ -133,13 +134,16 @@ ContactIdentityCompleter.prototype = {
       this.identityCollection =
         this.contactCollection.subCollections[Gloda.NOUN_IDENTITY];
 
-      let contactNames = [(c.name.replace(" ", "").toLowerCase() || "x") for each
-                          ([, c] in Iterator(this.contactCollection.items))];
+      let contactNames = this.contactCollection.items.map(function(c) {
+        return c.name.replace(" ", "").toLowerCase() || "x";
+      });
       // if we had no contacts, we will have no identity collection!
       let identityMails;
-      if (this.identityCollection)
-        identityMails = [i.value.toLowerCase() for each
-                         ([, i] in Iterator(this.identityCollection.items))];
+      if (this.identityCollection) {
+        identityMails = Array.prototype.map.call(this.identityCollection.items, function(i) {
+          return i.value.toLowerCase();
+        });
+      }
 
```

Comment:
why not map here? I suspect identityCollection is a glodaCollection too, meaning its items property is an array, meaning you can call map on it